### PR TITLE
Created note listing page

### LINKED ISSUE #5: Create note listing page

### PR DESCRIPTION
**Pull request by chiragbagde**

### CODE REVIEW

**Comment by yedhink** [comment]
@chiragbagde Go through https://handbook.neetokb.com/articles/git-and-github-essentials very carefully and see if you need to make any changes in this PR.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      style="secondary"
+    />
+    <div className="flex flex-row gap-2 text-gray-400">
+      <Clock />
+      <Typography style="body2">{createdAt}</Typography>
+      <Avatar
+        size="small"
+        user={{
+          imageUrl: "https://i.pravatar.cc/300",
+          name: "Neeraj",
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:20]: Don't use real names. Stick onto the personas defined at https://handbook.neetokb.com/articles/personas

File: app/javascript/src/components/Dashboard/Notes/Note/Tooltip.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import { Button } from "neetoui";
+
+const Tooltip = () => (
+  <Button
+    className="absolute right-12"
+    label="Wednesday 10:30AM"
+    onClick={() => {}}
+  />
+);
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Tooltip.jsx:11]: - The component name and the content doesn't match
- More than that this abstraction is pretty unnecessary. The content in this component doesn't justify the need for making it a separate component. It's very small and not too complex.
- Unless this component is reused at multiple places, it doesn't make sense to have this as a separate component.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line react/jsx-filename-extension
+import React from "react";
+
+import Content from "./Content.jsx";
+import Footer from "./Footer.jsx";
+import Header from "./Header.jsx";
+import Tooltip from "./Tooltip.jsx";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/index.jsx:7]: Go through the Slack thread.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const Note = ({ NOTES_DATA }) =>
+  NOTES_DATA.map(NOTE => (
+    <>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/index.jsx:11]: Why do we need this fragment?

File: app/javascript/src/components/Dashboard/Notes/Note/Content.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+import { Typography } from "neetoui";
+
+const Content = ({ text }) => (
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Content.jsx:5]: This needn't be a separate component. Refer prev comments.

File: app/javascript/src/components/Dashboard/Notes/NotesHeader.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+
+import { Button } from "neetoui";
+import { Header } from "neetoui/layouts";
+
+const NotesHeader = ({ setShowNewNotePane }) => {
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NotesHeader.jsx:6]: https://academy.bigbinary.com/learn-react/pages/structuring-code#:~:text=Avoid%20Unnecessary%20Namespacing

**Review by yedhink — CHANGES_REQUESTED** [review]
@chiragbagde

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 4 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx:11]: Since I wanted spacing between the individual note, I needed a break statement. To enclose individual components and break statement I used this fragment.

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx:11]: Got it. Didn't see the `br` tag.

**Comment by chiragbagde** [comment]
Made the required changes.

**Comment by chiragbagde** [comment]
Made the required changes.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx (hunk 1 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx]: Renamed it.

File: app/javascript/src/components/Dashboard/Notes/Note/Tooltip.jsx (hunk 2 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Tooltip.jsx]: Removed this component.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 3 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx]: Made the changes.

File: app/javascript/src/components/Dashboard/Notes/Note/Content.jsx (hunk 5 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Content.jsx]: Removed this component.

File: app/javascript/src/components/Dashboard/Notes/NotesHeader.jsx (hunk 6 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/NotesHeader.jsx]: Made the changes.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx [hunk 7]
```diff
@@ -1,19 +1,28 @@
...
+        <Typography className="mb-4 text-gray-400" style="body2">
+          {NOTE.text}
+        </Typography>
+        <hr className="mb-4" />
         <Footer createdAt={NOTE.createdAt} />
-        {NOTE.createdAt.includes("Created") && <Tooltip />}
+        {NOTE.createdAt.includes("Created") && (
+          <Button
+            className="absolute right-12"
+            label="Wednesday 10:30AM"
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/index.jsx:22]: What does this value represent? If it represents the creation date of the note, then use the `createdAt` prop of a `note` rather than hardcoding.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 8]
```diff
@@ -6,8 +6,8 @@ import { Container } from "neetoui/layouts";
 import { NOTES_DATA } from "./constants";
 import DeleteAlert from "./DeleteAlert";
 import Note from "./Note";
-import NotesHeader from "./NotesHeader";
 import NewNotePane from "./Pane/Create";
+import Header from "./TitleBar";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:10]: I don't understand the logic behind this naming. How do you associate TitleBar to Header? And most important question here is why not name the component and file as `Header` itself? Why name it `TitleBar` and then import it under an alias?

**Review by yedhink — CHANGES_REQUESTED** [review]
@chiragbagde

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 8 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:10]: Since the Header component already had a Header component from neetoui/layouts naming the main component also as Header was giving an error.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 7 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx:22]: The creation date of the note value for the individual component is displayed in the footer component This logic is just for showing the button with the status created.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 8 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:10]: @chiragbagde Read: https://academy.bigbinary.com/learn-react/pages/miscellaneous-tips#:~:text=Importing%20under%20an%20alias

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 7 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx:22]: @chiragbagde I am talking about the `label` of this Button. What does the date represent? Why that date is hardcoded?

**Comment by chiragbagde** [comment]
Fixed it.

**Review by yedhink — CHANGES_REQUESTED** [review]
@chiragbagde

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx [hunk 9]
```diff
@@ -6,6 +6,8 @@ import { Button, Typography } from "neetoui";
 import Footer from "./Footer";
 import Header from "./Header";
 
+import { displayDayAndTime, displayTimeFromCreation } from "../utils";
```
[the remark this answers is not in the document]
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx]: Renamed it.

**Comment by yedhink** [comment]
@chiragbagde Also make sure to acknowledge/respond to questions from someone else. Refer: https://github.com/chiragbagde/wheel/pull/6#discussion_r976336847

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 7 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx]: Sorry I missed replying to this comment.  I've added dayjs in the button component and shifted my hardcoded logic.

**Comment by chiragbagde** [comment]
I've updated all the changes.

File: app/javascript/src/components/Common/Sidebar/index.jsx [hunk 10]
```diff
@@ -46,23 +48,26 @@ const Sidebar = () => {
...
+        }}
+        profileInfo={{
+          name: `${user.first_name} ${user.last_name}`,
+          imageUrl: user.profile_image_path,
+          email: user.email,
+          bottomLinks,
+        }}
+        onCollapse={() => setIsSidebarCollapsed(!isSidebarCollapsed)}
+      />
+      <NotesSidebar />
```
Comment (yedhink) [inline app/javascript/src/components/Common/Sidebar/index.jsx:69]: Why do we have to keep the NotesSidebar in this component? So if we switch to the `Contact` page, then wouldn't we be seeing the NoteSidebar in that page too?

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+const Footer = ({ createdAt }) => (
+  <div className="flex flex-row items-center justify-between">
+    <Button
+      className="text-gray-400"
+      label="Getting Started"
+      style="secondary"
+    />
+    <div className="flex flex-row gap-2 text-gray-400">
+      <Clock />
+      <Typography style="body2">{createdAt}</Typography>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:15]: Should createdAt be displayed in the default format over here?

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import { Typography, Button, Avatar } from "neetoui";
+
+const Footer = ({ createdAt }) => (
+  <div className="flex flex-row items-center justify-between">
+    <Button
+      className="text-gray-400"
+      label="Getting Started"
+      style="secondary"
+    />
+    <div className="flex flex-row gap-2 text-gray-400">
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:13]: Why do we need to pass `text-gray-400` here?

File: app/javascript/src/components/Dashboard/Notes/Note/Header.jsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+import { MenuVertical } from "neetoicons";
+import { Typography, Dropdown } from "neetoui";
+
+const Header = ({ title }) => (
+  <div className="flex justify-between font-bold ">
+    <Typography style="h4">{title}</Typography>
+    <Dropdown buttonStyle="text" icon={MenuVertical}>
+      <li>Edit</li>
+      <li>Delete</li>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: Go through the Slack thread.

File: app/javascript/src/components/Dashboard/Notes/Note/Header.jsx [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+import { MenuVertical } from "neetoicons";
+import { Typography, Dropdown } from "neetoui";
+
+const Header = ({ title }) => (
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:6]: Why do we need to abstract this into a separate component? Go through the Slack thread.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+import { Button, Typography } from "neetoui";
+
+import Footer from "./Footer";
+import Header from "./Header";
+
+import { formatDateAndTime, calculateCreatedAgo } from "../utils";
+
+const Note = ({ NOTES_DATA }) =>
+  NOTES_DATA.map(NOTE => (
+    <>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Note/index.jsx:13]: Read: https://academy.bigbinary.com/learn-react/pages/using-unique-keys-for-list-elements

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 16]
```diff
@@ -44,47 +31,8 @@ const Notes = () => {
...
-        <EmptyState
-          image={EmptyNotesListImage}
-          primaryAction={() => setShowNewNotePane(true)}
-          primaryActionLabel="Add New Note"
-          subtitle="Add your notes to send customized emails to them."
-          title="Looks like you don't have any notes!"
-        />
-      )}
+      <Header setShowNewNotePane={setShowNewNotePane} />
+      <Note NOTES_DATA={notes} />
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:35]: This is not right. Here you have named the component as `Note`, singular. Which means it takes care of a single note. But you have passed in plural, `notes`. It doesn't make any sense. Refactor.

**Review by yedhink — CHANGES_REQUESTED** [review]
@chiragbagde

File: app/javascript/src/components/Common/Sidebar/index.jsx (hunk 10 above)
Comment (chiragbagde) [reply app/javascript/src/components/Common/Sidebar/index.jsx:69]: Initially I kept component in index.js. I thought of changing this while working on the Contacts Page. Now I've changed it.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx (hunk 11 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:15]: I've changed this format.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx (hunk 12 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:13]: I wanted to give light gray color for this footer component.

File: app/javascript/src/components/Dashboard/Notes/Note/Header.jsx (hunk 14 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:6]: Removed the separate components from Note.

File: app/javascript/src/components/Dashboard/Notes/Note/index.jsx (hunk 15 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/index.jsx:13]: Fixed it.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 16 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:35]: Made the changes.

File: app/javascript/src/components/Dashboard/Notes/Note/Header.jsx (hunk 13 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: @chiragbagde Not helpful. If you were told the same sentence that you have written, that is `NeetoUI Menuitem.Button is giving an error.`, then would you be able to easily figure out what's the issue?

Put up a video showing what's the error and when it occurs. Also point out your neetoUI version.

Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: Sorry the message was vague and not properly communicated from my end. What I meant was when I try to render Dropdown from neetoui I was getting an error. MenuItem.Button from dropdown was giving this error. I have pointed out the issue in this video.
[https://www.loom.com/share/c4da54cb63cf40bfa50656c170072f1d](url)

Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: Initially I was also using docker. Later I tried bundle install but its giving me an error while installation similar to this video [https://www.loom.com/share/6378c61cfcef4bfdb92fcd63ef06cc41](url).  I tried the solution mentioned here but still its giving the same error.
error-2
-55344ebafaea.png">

Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: Initially I was also using docker. Now when I try bundle install I'm getting an error while installation as 'Gem::Ext::BuildError: ERROR: Failed to build gem
native extension'.I tried the solution mentioned in slack video but it’s still giving the same error.
error-1

error-2

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: >Initially I was also using docker. Now when I try bundle install I'm getting an error while installation as 'Gem::Ext::BuildError: ERROR: Failed to build gem
native extension'.I tried the solution mentioned in slack video but it’s still giving the same error. The error is occurring while installing ffi package.

Does this section resolve the issue: https://www.bigbinary.com/books/learn-rubyonrails-book/installing-ruby-on-rails#typeerror-for-rails-project-in-m1-mac ?

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: >MenuItem.Button from dropdown was giving this error.

@chiragbagde Regarding the `MenuItem`. Refer the steps: https://github.com/Ishaan-Singhal/neetoUI-challenge/pull/9#discussion_r978600442

Also ensure you have restarted your Rails as well as webpack servers once you upgrade the versions.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx (hunk 12 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:13]: @chiragbagde https://www.loom.com/share/480e134db41446498eee7ae3155255ec

File: app/javascript/src/components/Dashboard/Notes/Note/Header.jsx (hunk 13 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Header.jsx:11]: Now my server is running using bundle install and it also displays the MenuItem Button.

File: app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx (hunk 12 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/Note/Footer.jsx:13]: When I click on this property in the Figma design for this challenge, it displays the CSS code as font-weight 400 with light grey.

**Comment by chiragbagde** [comment]
I have made the required changes.

**Comment by chiragbagde** [comment]
I have made the required changes.

File: app/javascript/src/components/Dashboard/Notes/List.jsx [hunk 17]
```diff
@@ -0,0 +1,57 @@
+// eslint-disable-next-line react/jsx-filename-extension
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/List.jsx:1]: Why is this required here? Remove these comments which are disabling eslint rules. Unless your consult with the tech lead you should not be adding these comments.

**Review by yedhink — COMMENTED** [review]
@chiragbagde

File: app/javascript/src/components/Dashboard/Notes/List.jsx (hunk 17 above)
Comment (chiragbagde) [reply app/javascript/src/components/Dashboard/Notes/List.jsx:1]: Removed it.
